core/jobs.py

```diff
@@ -205,13 +205,13 @@ def register_completion(cls, job_id, output_list):
 
         model.status_code = STATUS_CODE_COMPLETED
         model.time_finished_msec = utils.get_current_time_in_millisecs()
-        model.output = cls._compress_output_list(output_list)
+        model.output = cls.compress_output_list(output_list)
         model.put()
 
         cls._post_completed_hook(job_id)
 
     @classmethod
-    def _compress_output_list(
+    def compress_output_list(
```

Comment:
Nope, this isn't what we want to do, this should remain private. You want to make a wrapper for this.

Reply:
I have created wrapper function for this method

Reply:
Actually, tests shouldn't drive changes to the API that a class exposes. I.e. if a method is meant to be private then it shouldn't be switched to a public one just for the purposes of tests. Adding a public wrapper for a private method is just a roundabout way of doing that, and it can cause confusion because there is now a private method and a public method and no one is sure which one to use.

So, I think there are two choices:

(a) Make the existing method public. This is not so great if it's not actually needed in the public API, but it might be necessary to do that if (b) is too hard.

(b) Find a way to test the code in the private method through a public method (register_completion in this case). This would involve moving the optional test_only_* arg to the public register_completion method and only calling register_completion in the tests, in such a way that the code in the private method continues to be well-tested.

I think (b) should be feasible -- perhaps you can try it and let us know what you find?